apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/PipelineManager.java

```diff
@@ -611,8 +593,19 @@ public void onFailure(Throwable jobThrowable) {
          * since it won't get called in the first place).
          */
         handleJobCancellation(jobRecord.getId());
+        LOGGER.info("Job cancelled: " + jobRecord.getJobType());
+      } else if (jobThrowable instanceof InterruptedException) {
+        // If our job has been interrupted, we are already shutting down, so just ignore it.
+        LOGGER.info("Job interrupted: " + jobRecord.getJobType());
+      } else {
+        /* If the job failed and wasnt cancelled, it threw some other exception mid-stream,
+         * and now we must die as we almost certainly cannot continue with the tainted batch sitting in Incoming.
+         * This will call the proper shutdown procedures and exit gracefully.
+         * FUTURE: We may be able to more fault-tolerant by moving the failed batch
+         * into the Failed folder when this happens and continuing to run
+         */
+        PipelineApplication.shutdown();
```

Comment:
Here we fix a problem where originally this simply logged the error and expected the original exception to bubble back to PipelineApplication for initiating a shutdown, which wasnt happening due to threading. Now we explicitly shut down the pipeline.

Reply:
Will the exception be logged somewhere else?

Reply:
Will other jobs have an opportunity to shutdown cleanly?  This looks like it will just terminate the app instantly.

Reply:
Yes the original exception is logged before it gets here. This will call PipelineApplicaton which will call System.exit(), which then runs the shutdown routines in PipelineManager (and interrupts any ongoing threads that allow themselves to be interrupted). PipelineApplication will wait for its jobs to either finish (if uninterruptible like RifLoader) or cancel them (if they havent started running or are interruptible.)